File: Java/0001-1000/001-100/21-30/21.java
/**
 * Definition for singly-linked list. public class ListNode { int val; ListNode
 * next; ListNode() {} ListNode(int val) { this.val = val; } ListNode(int val,
 * ListNode next) { this.val = val; this.next = next; } }
 */
class Solution {
    public ListNode mergeTwoLists(ListNode l1, ListNode l2) {
        ListNode l3 = new ListNode(), l4 = l1, l5 = l2, l6 = new ListNode(), cur = l3;
        while (l4 != null && l5 != null) {
            if (l4.val < l5.val) {
                l6 = new ListNode(l4.val);
                l4 = l4.next;
            } else {
                l6 = new ListNode(l5.val);
                l5 = l5.next;
            }
            cur.next = l6;
            cur = l6;
        }
        while (l4 != null) {
            l6 = new ListNode(l4.val);
            l4 = l4.next;
            cur.next = l6;
            cur = l6;
        }
        while (l5 != null) {
            l6 = new ListNode(l5.val);
            l5 = l5.next;
            cur.next = l6;
            cur = l6;
        }
        return l3.next;
    }
}

File: Python/0001-1000/001-100/21-30/21.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeTwoLists(self, l1: ListNode, l2: ListNode) -> ListNode:
        ret = ListNode()
        now = ret
        while (not l1 is None) and (not l2 is None):
            if l1.val < l2.val:
                x = ListNode(l1.val)
                l1 = l1.next
            else:
                x = ListNode(l2.val)
                l2 = l2.next
            now.next = x
            now = now.next

        while not l1 is None:
            x = ListNode(l1.val)
            l1 = l1.next
            now.next = x
            now = now.next

        while not l2 is None:
            x = ListNode(l2.val)
            l2 = l2.next
            now.next = x
            now = now.next

        return ret.next
